source/slang/slang-emit-c-like.cpp

```diff
@@ -4168,11 +4217,13 @@ void CLikeSourceEmitter::emitGlobalParam(IRGlobalParam* varDecl)
     SLANG_ASSERT(layout);
 
     emitVarModifiers(layout, varDecl, varType);
+    emitVarAttributes(varType, layout);
```

Comment:
I think we can just use `emitVarModifiers` for this purpose.

Reply:
Should I make `emitVarModifiers` virtual and then override the whole thing? It's currently not virtual and contains a bunch of calls to virtual functions. I'm not sure which ones of those would fit in this case. 
I guess `emitVarDecorationsImpl` could be close, but it just takes the `decl` so I'd have to replicate some code to get the layout and type as well (or just add those as arguments).

Reply:
will `emitLayoutQualifiersImpl` work?

Reply:
> will `emitLayoutQualifiersImpl` work?

Not currently because that only gets passed the layout, and I expect that I will also need the type:
(Yes I know this code is very wrong, but eventually I need to output "set" and "binding" here, and I need the type for that. For now I just need some code to emit distinct "set" and "binding" so that the output at least compiles.)


    void WGSLSourceEmitter::emitVarAttributesImpl(IRType * type, IRVarLayout* layout)
    {
        for (auto attr : layout->getOffsetAttrs())
        {
            // TODO:
            // This is not correct. For the moment this is just here as a hack to make
            // @binding and @group unique, so that we can pass WGSL compile tests.
            // This will have to be revisited when we actually want to supply resources to
            // shaders.
            if (
                (type->getOp() == kIROp_HLSLRWStructuredBufferType) ||
                (type->getOp() == kIROp_HLSLStructuredBufferType)
                )
            {
                m_writer->emit("@binding(");
                m_writer->emit(attr->getOffset());
                m_writer->emit(") ");
                m_writer->emit("@group(");
                m_writer->emit(attr->getSpace());
                m_writer->emit(") ");
                return;
            }
        }
    }

Reply:
Instead of checking types, each offset attr provides `getResourceKind()`, and you can use that to decide whether or not to emit these decorations.